crypto/asn1/x_long.c

```diff
@@ -49,13 +49,13 @@ ASN1_ITEM_end(ZLONG)
 
 static int long_new(ASN1_VALUE **pval, const ASN1_ITEM *it)
 {
-    *(long *)pval = it->size;
+    memcpy(pval, &it->size, sizeof(it->size));
```

Comment:
It should be the minimum of sizeof(src) and sizeof(dest), a long is being copied into a pointer and either could be larger.